cmd/object-handlers.go

```diff
@@ -171,29 +171,23 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 		}
 	}
 
-	setObjectHeaders(w, objInfo, hrange)
 	setHeadGetRespHeaders(w, r.URL.Query())
-	httpWriter := ioutil.WriteOnClose(writer)
+	setObjectHeaders(w, objInfo, hrange)
+
+	// HTTP status code is returned at this stage: no error should
+	// be returned at this stage.
 
 	getObject := objectAPI.GetObject
 	if api.CacheAPI() != nil && !hasSSECustomerHeader(r.Header) {
 		getObject = api.CacheAPI().GetObject
 	}
 
-	// Reads the object at startOffset and writes to mw.
-	if err = getObject(ctx, bucket, object, startOffset, length, httpWriter, objInfo.ETag); err != nil {
-		if !httpWriter.HasWritten() { // write error response only if no data has been written to client yet
```

Comment:
This code was to detect if we started writing if we can keep this and have `httpWriter := ioutil.WriteOnClose(writer)`  but we should detect that we have written atleast the part about http status WriteOnClose()